iothub/device/src/IotHubBaseClient.cs

```diff
@@ -60,14 +63,28 @@ public abstract class IotHubBaseClient : IAsyncDisposable
                 ModelId = _clientOptions.ModelId,
                 PayloadConvention = _clientOptions.PayloadConvention,
                 IotHubClientTransportSettings = _clientOptions.TransportSettings,
-                HttpOperationTransportSettings = _clientOptions.FileUploadTransportSettings,
                 MethodCallback = OnMethodCalledAsync,
                 DesiredPropertyUpdateCallback = OnDesiredStatePatchReceived,
                 ConnectionStatusChangeHandler = OnConnectionStatusChanged,
                 MessageEventCallback = OnMessageReceivedAsync,
                 RetryPolicy = _clientOptions.RetryPolicy ?? new IotHubClientNoRetry(),
             };
 
+            if (IotHubConnectionCredentials.ModuleId.IsNullOrWhiteSpace())
+            {
+                // Set up file upload settings over HTTP for the device client
+                PipelineContext.HttpOperationTransportSettings = _clientOptions.FileUploadTransportSettings;
+            }
+            else
+            {
+                // Set the remote certificate validator for the module client
+                _certValidator = certificateValidator ?? NullCertificateValidator.Instance;
+                PipelineContext.HttpOperationTransportSettings = new IotHubClientHttpSettings
+                {
+                    ServerCertificateCustomValidationCallback = _certValidator.GetCustomCertificateValidation()
+                };
+            }
```

Comment:
Set up the mutually exclusive HTTP settings for device clients and module clients.
Device clients - file upload
Module client - edge module-edge device/ edge module methods